rfcs/0159-general-purpose-allocator-module.md

```diff
@@ -0,0 +1,151 @@
+---
+feature: general-purpose-allocator-module
+start-date: 2023-08-11
+author: lucasew
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+A function that generates a suggestion based item allocator module.
+
+# Motivation
+[motivation]: #motivation
+
+Sometimes there are some values that the user don't actually care about which
+value a option will get in some kind of space as long is a valid one and doesn't
+conflict with other definitions.
+
+One of these spaces, for example, is the port space, like, non administrative ports
+for servers (1025..49151). You will probably put a reverse proxy in front of it
+so even the stability of the port number is not so important.
+
+# Detailed design
+[design]: #detailed-design
```

Comment:
In ports space it might be interesting to have sub-spaces (ranges of allocation?), where a some services could be associated with a range of possible ports. For example my service could have a public API on port 7000 and a more restricted API on port 7001. Both are related to the same service and could be grouped together.

These sub-spaces would ultimately be flattened out and checked like any other values in that space, but would allow grouping of related values.

Reply:
https://github.com/lucasew/nixcfg/blob/2a153c7e4b131f0d3bb0e80ab8248fec4f57c15d/nix/nodes/common/services/transmission.nix#L10